cluster_test.go

```diff
@@ -683,6 +683,40 @@ var _ = Describe("ClusterClient", func() {
 			Expect(len(keys)).To(BeNumerically("~", nkeys, nkeys/10))
 		})
 
+		It("retries pipeline network errors", func() {
+			optNoRetry := redisClusterOptions()
+			optNoRetry.MaxRedirects = 1
+			clientNoRetry := cluster.clusterClient(optNoRetry)
+			defer clientNoRetry.Close()
+			// Set the read timeout after initializing the nodes
+			optNoRetry.ReadTimeout = time.Nanosecond
+
+			optRetry := redisClusterOptions()
+			optRetry.MaxRedirects = 5
+			optRetry.MaxRetryBackoff = 128 * time.Millisecond
+			clientRetry := cluster.clusterClient(optRetry)
+			defer clientRetry.Close()
+			optRetry.ReadTimeout = time.Nanosecond
+
+			startNoRetry := time.Now()
+			_, err := clientNoRetry.Pipelined(func(pipe redis.Pipeliner) error {
+				pipe.Ping()
+				return nil
+			})
+			Expect(err).To(HaveOccurred())
+			elapseNoRetry := time.Since(startNoRetry)
+
+			startRetry := time.Now()
+			_, err = clientRetry.Pipelined(func(pipe redis.Pipeliner) error {
+				pipe.Ping()
+				return nil
+			})
+			Expect(err).To(HaveOccurred())
+			elapseRetry := time.Since(startRetry)
+
+			Expect(elapseRetry).To(BeNumerically(">", elapseNoRetry, 10*time.Millisecond))
```

Comment:
Well, this is not very reliable. Yes, clientRetry does more work than clientNoRetry but that is the only thing we test. No guarantees it does anything useful...

Reply:
I followed the pattern used in the `should retry with backoff` test (https://github.com/go-redis/redis/blob/4b568cdf1adcfe5cab06ad6929ad5286b3db7138/redis_test.go#L158). Can you suggest a better way of testing it?

Reply:
No, not really - sorry.